spec2sdk/parsers/resolver.py

```diff
@@ -68,13 +68,14 @@ def resolve_reference(reference: str) -> dict:
 
             return self._resolved_references[reference_id]
 
-        return {
-            new_key: new_value
-            for key, value in schema.items()
-            for new_key, new_value in (
-                resolve_reference(value) if key == "$ref" else {key: resolve_value(value)}
-            ).items()
-        }
+        # Object containing $ref cannot be extended with additional properties
+        # and any properties added SHALL be ignored.
+        # https://spec.openapis.org/oas/v3.0.0.html#reference-object
+        return (
+            resolve_reference(schema["$ref"])
+            if "$ref" in schema
```

Comment:
Do you want to ignore any extra properties silently or throw an exception when that happens?

Reply:
I want to ignore extra properties. It's validator responsibility to raise an exception. Unfortunately, in our case validator runs after JSON schema resolving. This should be fixed in the future.